SERVER/src/checkpoint_manager.py

```
@@ -1,105 +1,115 @@
+
+            # 2. Restore the server's LLM adapters
+            adapters_in_checkpoint = os.path.join(checkpoint_dir, f"{self.server_Actor_id}_adapters")
+            if os.path.exists(adapters_in_checkpoint):
+                if os.path.exists(self.server_adapter_specific_path): # Use SERVER_ADAPTER_SPECIFIC_PATH
+                    shutil.rmtree(self.server_adapter_specific_path) # Remove existing adapters before copying
+                shutil.copytree(adapters_in_checkpoint, self.server_adapter_specific_path) # Use SERVER_ADAPTER_SPECIFIC_PATH
+            else:
+                print(f"Warning: No {self.server_Actor_id} adapters found in checkpoint '{checkpoint_name}'. Skipping adapter load.")
+
+            return f"Checkpoint '{checkpoint_name}' loaded. PLEASE RESTART THE APPLICATION for changes to take effect."
+        except Exception as e:
```

Ruff (0.11.9)

Do not catch blind exception: `Exception`

(BLE001)